Author: Robert James Clay <jame@rocasa.us>
Description: Set working directory in the apache-vhost.conf, nginx-vhost.conf,
 and lighttpd-vhost.conf http proxy configuration files and otherwise adapt them
 for use on Debian systems. Correct the default backend port in the example 
 default.vcl file for Varnish. Also reference default SSL certificates using the 
 so-called snakeoil ones as working examples.
Origin: vendor
Last-Update: 2019-07-04
--- a/doc/conf/apache-vhost.conf
+++ b/doc/conf/apache-vhost.conf
@@ -20,15 +20,15 @@
 <VirtualHost *:443>
   ServerName YOUR_SERVER_NAME
 
-  DocumentRoot WORKING_DIR/UI
+  DocumentRoot /usr/share/ledgersmb-1.7/UI
 
   # If you own a publicly exposed server, consider submitting it
   # to the SSL security tests available at
   #    https://www.ssllabs.com/ssltest/
   SSLEngine On
-  SSLCertificateFile SSL_CERT_FILE
-  SSLCertificateKeyFile SSL_KEY_FILE
-  SSLCertificateChainFile SSL_CHAIN_FILE
+  SSLCertificateFile /etc/ssl/certs/ssl-cert-snakeoil.pem
+  SSLCertificateKeyFile /etc/ssl/private/ssl-cert-snakeoil.key
+  #SSLCertificateChainFile SSL_CHAIN_FILE
 
   <Location "/">
     SSLRequireSSL
--- a/doc/conf/nginx-vhost.conf
+++ b/doc/conf/nginx-vhost.conf
@@ -22,12 +22,12 @@
   # Replace snippets/snakeoil.conf with either your own version of snakeoil.conf
   # or modified versions of the following ssl_certificate, ssl_certificate_key lines
   include snippets/snakeoil.conf;
-  #ssl_certificate SSL_CERT_FILE;
+  ssl_certificate /etc/ssl/certs/ssl-cert-snakeoil.pem;
   # after expansion, the above may look like:
   # ssl_certificate /etc/certs/example.com.pem;
-  #ssl_certificate_key SSL_KEY_FILE;
+  #ssl_certificate_key /etc/ssl/private/ssl-cert-snakeoil.key;
 
-  root WORKING_DIR/UI;
+  root /usr/share/ledgersmb-1.7/UI;
 
   # Configuration files don't exist
   location ^~ \.conf$ {
--- a/doc/conf/lighttpd-vhost.conf
+++ b/doc/conf/lighttpd-vhost.conf
@@ -10,19 +10,11 @@
 
 server.modules = (
 	"mod_expire",
-	"mod_access",
-	"mod_alias",
-	"mod_compress",
- 	"mod_redirect",
 )
 
 server.name YOUR_SERVER_NAME;
 
-server.document-root        = "WORKING_DIR/UI"
-server.errorlog             = "/var/log/lighttpd/error.log"
-server.pid-file             = "/var/run/lighttpd.pid"
-server.username             = "www-data"
-server.groupname            = "www-data"
+server.document-root        = "/usr/share/ledgersmb-1.7/UI"
 
   # If you own a publicly exposed server, consider submitting it
   # to the SSL security tests available at
@@ -31,27 +23,14 @@
   # Replace snippets/snakeoil.conf with either your own version of snakeoil.conf
   # or modified versions of the following ssl_certificate, ssl_certificate_key lines
   include snippets/snakeoil.conf;
-  #ssl_certificate SSL_CERT_FILE;
+  ssl_certificate /etc/ssl/certs/ssl-cert-snakeoil.pem;
   # after expansion, the above may look like:
   # ssl_certificate /etc/certs/example.com.pem;
-  #ssl_certificate_key SSL_KEY_FILE;
+  ssl_certificate_key /etc/ssl/private/ssl-cert-snakeoil.key;
 
 server.port                 = 443
 ssl.engine                  = "enable"
-ssl.pemfile                 = "SSL_CERT_FILE"
-
-index-file.names            = ( "login.pl", "index.php", "index.html", "index.lighttpd.html" )
-url.access-deny             = ( "~", ".inc" )
-static-file.exclude-extensions = ( ".php", ".pl", ".fcgi" )
-
-compress.cache-dir          = "/var/cache/lighttpd/compress/"
-compress.filetype           = ( "application/javascript", "text/css", "text/html", "text/plain" )
-
-# default listening port for IPv6 falls back to the IPv4 port
-## Use ipv6 if available
-include_shell "/usr/share/lighttpd/use-ipv6.pl " + server.port
-include_shell "/usr/share/lighttpd/create-mime.assign.pl"
-include_shell "/usr/share/lighttpd/include-conf-enabled.pl"
+ssl.pemfile                 = "/etc/ssl/certs/ssl-cert-snakeoil.pem"
 
 server.modules   += ( "mod_proxy" )
 
--- a/doc/conf/default.vcl
+++ b/doc/conf/default.vcl
@@ -15,7 +15,7 @@
 # Default backend definition. Set this to point to your content server.
 backend default {
     .host = "127.0.0.1";
-    .port = "5000";
+    .port = "5762";
 }
 
 sub vcl_recv {
